pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -171,6 +171,21 @@ func (rs *RevisionStatus) PropagateDeploymentStatus(original *appsv1.DeploymentS
 func (rs *RevisionStatus) PropagateAutoscalerStatus(ps *autoscalingv1alpha1.PodAutoscalerStatus) {
 	// Reflect the PA status in our own.
 	cond := ps.GetCondition(autoscalingv1alpha1.PodAutoscalerConditionReady)
+
+	if actualScale := ps.GetActualScale(); actualScale != -1 {
+		rs.ActualReplicas = actualScale
+	} else {
+		rs.ActualReplicas = 0
+	}
+	//rs.ActualReplicas = ps.GetActualScale()
+
+	rs.DesiredReplicas = ps.GetDesiredScale()
+	if desiredScale := ps.GetDesiredScale(); desiredScale != -1 {
+		rs.DesiredReplicas = desiredScale
+	} else {
+		rs.DesiredReplicas = 0
+	}
```

Comment:
Can we tidy up this section to so the two `if` blocks are similar

Reply:
sure, I must have messed up my rebase sorry about that